src/Evolution/Executables/GeneralizedHarmonic/EvolveGhBinaryBlackHole.hpp

```diff
@@ -180,6 +180,28 @@ namespace Parallel {
 template <typename Metavariables>
 class CProxy_GlobalCache;
 }  // namespace Parallel
+
+// We don't use the Tags::TimeAndPreviousCompute in
+// Time/Tags/TimeAndPrevious.hpp because it uses the
+// ::evolution::Tags::PreviousTriggerTime tag and that tag is set to NaN outside
+// of the dense triggers. This is going to be used in the observation horizon
+// finds which are typically run on standard triggers so using
+// ::evolution::Tags::PreviousTriggerTime would cause an FPE. For the
+// observation horizon finds, we don't care about the previous time because
+// it'll never be used so we just set it to nullopt.
```

Comment:
This sounds like a workaround for a bug in the horizon finder.  Why would you ignore the second field in a `LinkedMessageId`?  That's the whole point of the class.  It's like using an atomic type and then never performing any atomic operations on it.

Reply:
This is precisely a workaround for a _feature_ of the horizon finder. I don't think trying to explain it here is a good idea, so let's talk about it at the Monday call.